torch/lib/c10d/NCCLUtils.hpp

```diff
@@ -85,27 +87,31 @@ class NCCLComm {
 
   // Move constructable
   NCCLComm(NCCLComm&& other) {
```

Comment:
You need to grab the mutex of the other NCCLComm because `this` is being constructed.